docs/workflows/best-practices-osh.md

```diff
@@ -0,0 +1,398 @@
+> this also applies to frameworks
+> often, libraries are part of an ecosystem, or work within a certain application framework; e.g. eclipse, Apache, where it makes a lot of sense (consistency, frictionless compatibility) to pick a library from the same ecosystem, when that meets the necessary requirements. 
+- checklist:
+  - Does it have open issues?​
+  - Is the license acceptable?​
+    - Libraries must have a license acceptable to <org>.  maintains a list of common licenses used in free and open-source software (FOSS); if a library has a license listed as acceptable, it can be used. 
+  - How is the code quality of the library? 
+  - Does it contain optimizations for performance, maintainability, functionality?​
+  - Is the new version compatible?  ​
+  - How mature is the version?​
+  - Is it stable enough?​
+  - Are many people using it (e.g. check GitHub stars)
+
+
+### 11. When a library does not support requirements
+There are several possible reasons why a library does not support the needs and requirements:
+1. _Functional mismatch_: a library is missing features that cannot easily be added on top, or the implementation of the library is based on assumptions or choices that are incompatible with the ones in the application.
+2. _Bug in library implementation_: typically detected after a library has been adopted, so the cost of switching is non-neglible.
+3. _Compatibility break_: a library does not (or no longer) work well together with another library or the application itself, due to changes in any of the involved components.
+
+The basic rule is that _library implementations should not be modified or customized_: ​One of the main benefits of libraries and frameworks, is that they provide functionality without the duty of maintaining it. After customizing a library implementation, you lose this benefit while being dependent on the changes that the community makes.​
+
+- First, check whether a newer version of the library may solve the issue, then consider updating ([Updating a library]); you may also wait a bit until a fix is being released, especially when the issue is being worked on.
+- If the issue is a bug, you can look at the code of the library and develop a bug fix: 
+  - You may be able to wrap the relevant library call(s) with extra code that corrects or hides the bug.
+  - Alternatively, you can temporarily use the modified library while merging back the bug fix into the library (be aware of the license). Once the community has accepted the fix, you can update and remove the local code​.
+
+- If no other solution is feasible and a modification is absolutely required (or: the costs of alternative solutions are very high), the source code should be 'adopted' (if permitted by its license) and put into a designated area in a version control system. The modification should be documented so that it can be re-applied whenever a newer version of the library is made available.
+
+### 12. Replacing a library
```

Comment:
I feel there's a bit of redundancy in the structure. To me, it appears that we have the following:
- Sometimes, you need to switch libraries. This can be caused by insufficient health (license, activity, management), compatibility issues or functional requirements.
- Then, this is how you go about it (Replacing a new library). When it's not a replacement but a new library, do the same checks minus a couple that only apply to a replacement. (where is the old one used and how is the interface different?)

Reply:
correct, I think this is better now